fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -1781,14 +1787,80 @@ public void onCreate() {
     }
 
     @Override
-    public void onDrop() {
+    public void onDrop(Database db, boolean force, boolean replay) {
         // drop all temp partitions of this table, so that there is no temp partitions in recycle bin,
         // which make things easier.
         dropAllTempPartitions();
     }
 
+    @Override
+    public Runnable delete(long dbId, boolean replay) {
+        GlobalStateMgr.getCurrentState().getLocalMetastore().onEraseTable(dbId, this, replay);
+        return replay ? null : new DeleteTableTask(this);
+    }
+
     @Override
     public boolean isSupported() {
         return true;
     }
+
+    private static class DeleteTableTask implements Runnable {
+        private final OlapTable table;
+
+        public DeleteTableTask(OlapTable table) {
+            this.table = table;
+        }
+
+        @Override
+        public void run() {
```

Comment:
This method was extracted from the original method `LocalMetaStore.onEraseOlapOrLakeTable()`